presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -166,6 +168,12 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             }
         }
 
+        if (config.getAwsAccessKey().isPresent() && config.getAwsSecretKey().isPresent()) {
+            AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(config.getAwsAccessKey().get(), config.getAwsSecretKey().get()));
+            asyncGlueClientBuilder.setCredentials(credentialsProvider);
+        }
+
```

Comment:
One more thing -- to actually achieve the desired effect that access keys "win" over IAM role, the `if (config.getIamRole().isPresent())` block needs to be changed into `else if (config.getIamRole().isPresent())`
(then also remove this blank space line)

Reply:
Fixed. 👍